app/src/main/java/org/mozilla/focus/fragment/DownloadDialogFragment.kt

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.fragment
+
+import android.os.Build
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import android.support.v7.app.AlertDialog
+import android.text.Html
+import android.text.Spanned
+import android.widget.Button
+import kotlinx.android.synthetic.main.download_dialog.view.*
+import mozilla.components.support.utils.DownloadUtils
+import org.mozilla.focus.R
+import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.web.Download
+
+/**
+ * Fragment displaying a download dialog
+ */
+class DownloadDialogFragment : DialogFragment() {
+
+    override fun onCreateDialog(bundle: Bundle?): AlertDialog {
+        val fileName = arguments!!.getString("fileName")
```

Comment:
On second thought, it seems if this fragment has no arguments, that's a red flag that should be caught. This is okay to crash here because we're expecting you should not use the DialogFragment without the newInstance() method.